pkg/operator/controller/ingress/status.go

```diff
@@ -188,10 +219,15 @@ func computeIngressDegradedCondition(conditions []operatorv1.OperatorCondition)
 			status:    operatorv1.ConditionTrue,
 		},
 		{
-			condition:   IngressControllerDeploymentDegradedConditionType,
-			status:      operatorv1.ConditionFalse,
+			condition:   IngressControllerDeploymentAvailableConditionType,
+			status:      operatorv1.ConditionTrue,
 			gracePeriod: time.Second * 30,
 		},
+		{
+			condition:   IngressControllerDeploymentReplicasAvailableConditionType,
+			status:      operatorv1.ConditionTrue,
+			gracePeriod: time.Second * 60,
```

Comment:
I'm contemplating adding another condition, `DeploymentMinReplicasAvailable`, which would be `True` iff `deployment.Status.AvailableReplicas < *deployment.Spec.Replicas - maxUnavailable`, where `maxUnavailable` would be computed using `deployment.Spec.Strategy.RollingUpdate.MaxUnavailable` (i.e., the ingresscontroller's `DeploymentMinReplicasAvailable` condition would report what the deployment's `Available` condition is supposed to report).  `DeploymentMinReplicasAvailable` would be included here in `expectedConditions` with a short grace period because its status should remain `True` during upgrades.  The grace period for `DeploymentReplicasAvailable` should be longer to allow it to become temporarily `False` during upgrades without causing the ingresscontroller to set `Degraded=True`.

Reply:
Would `DeploymentMinReplicasAvailable` be true if `deployment.Status.AvailableReplicas` were `0` and `*deployment.Spec.Replicas - maxUnavailable` was > 0, or is that not possible?

Reply:
> Would `DeploymentMinReplicasAvailable` be true if `deployment.Status.AvailableReplicas` were `0` and `*deployment.Spec.Replicas - maxUnavailable` was > 0, or is that not possible?

In that case, `DeploymentMinReplicasAvailable` would be `False`, as would be `DeploymentAvailable` and `DeploymentAvailable`.

Reply:
It might make sense to rename `DeploymentReplicasAvailable` to `DeploymentAllReplicasAvailable` as well.

Reply:
> It might make sense to rename `DeploymentReplicasAvailable` to `DeploymentAllReplicasAvailable` as well.

I am in favor of this, since it helps remove some of the ambiguity.